graphix/random_objects.py

```diff
@@ -0,0 +1,173 @@
+import numpy as np
+import numpy.typing as npt
+import scipy.linalg
+
+from graphix.channels import KrausChannel
+from graphix.sim.density_matrix import DensityMatrix
+from graphix.ops import Ops
+
+
+def rand_herm(l: int):
+    """
+    generate random hermitian matrix of size l*l
+    """
+    tmp = np.random.rand(l, l) + 1j * np.random.rand(l, l)
+    return tmp + tmp.conj().T
+
+
+def rand_unit(l: int):
+    """
+    generate random unitary matrix of size l*l from hermitian matrix
+    """
+    return scipy.linalg.expm(1j * rand_herm(l))
```

Comment:
How about using `scipy.stats.unitary_group`?
https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.unitary_group.html

Reply:
Used it in the beginning then tried to remove it to avoid using `scipy` but couldn't and used this more simple version. I opted for the easy version since I had no specific requirement on the measure used for the sampling but we might want to make sure that we use the Haar measure.

Reply:
I've read that Haar-random Unitary can be created using QR decomposition. Original paper is [1] and it is also cited in `scipy.stats.unitary_group` module. So, I believe it would be better to use `unitary_group` module!

[1] Francesco Mezzadri [How to generate random matrices from the classical compact groups](https://arxiv.org/abs/math-ph/0609050)
[2] [Understanding the Haar measure](https://pennylane.ai/qml/demos/tutorial_haar_measure/#mezzadri2006)

Reply:
As you want. I'm not sure it matters that much anyways.